openwisp_radius/utils.py

```diff
@@ -130,27 +130,35 @@ def find_available_username(username, users_list, prefix=False):
     return tmp
 
 
-def decode_csv_data(csv_data):
-    if isinstance(csv_data, bytes):
-        # Detect encoding using chardet
-        detected = chardet.detect(csv_data)
-        detected_encoding = detected.get('encoding')
-
-        # Explicit handling for UTF-16 encodings (check for BOM)
-        if csv_data.startswith(b'\xff\xfe') or csv_data.startswith(b'\xfe\xff'):
-            detected_encoding = 'utf-16'
-        detected_encoding = detected_encoding or 'utf-8'
+
+def get_encoding_format(csv_data):
+    
+    # Explicit handling for UTF-16 encodings (check for BOM)
+    if csv_data.startswith(b'\xff\xfe') or csv_data.startswith(b'\xfe\xff'):
+        return 'utf-16'
+
+    # Detect encoding using chardet
+    detected = chardet.detect(csv_data)
+    detected_encoding = detected.get('encoding')
+
+    if detected_encoding == 'ascii': 
+        return 'utf-8'
+    
+    if detected_encoding == 'utf-16le':
         try:
-            return csv_data.decode(detected_encoding)
-        except UnicodeDecodeError as e:
-            raise
-        return csv_data
+            csv_data.decode('utf-16le')  # Test if decoding works
```

Comment:
why test only for this encoding and not other encodings?

Reply:
i have explicilty already handled (uft-8, utf-16, ascii), utf-16le is the format which was not detected by the chardet library (many time chardet library return utf-8 instead of utf-16le), so i just trying in the end of the function, if it's utf-16le.